### ISSUE TITLE
Support Vinyl files stream as replacement input

### ISSUE THREAD

**Issue by jraoult** [VFK/gulp-html-replace#25]
I have this case where I'm generating a stylesheet from SASS and want to inject it into the page: I'm inline a subset of the main stylesheet for rendering optimisation (above the fold).

Right now, `gulp-html-replace` supports string or file path as a replacement but it would be nice for my use case, and any input actually, if it could be delivered through a stream of Vinyl files.

See bellow. The main stream would "wait" internally for the `cssInline` input stream to end then continue.

```
gulp.src('app/index.html', {base: 'app'})
        .pipe(htmlreplace({
          cssInline: {
            src: gulp.src('inline.scss').pipe(sass()),
            tpl: '<style>%s</style>'
          }
        }))
```

**Comment by jraoult** [comment]
btw, I'm happy to create a PR for that if you guys think it is a good idea :-)

**Comment by VFK** [comment]
Hi @jraoult it looks like a good idea, i'd love to see the PR. Also if you could update the readme according to the new functionality it'll be even more awesome :-) Thank you.

**Comment by jraoult** [comment]
Alright @VFK, I'll give a try asap. I'll keep posted.

**Comment by rafaelbiten** [comment]
+1
 
Yep, this would be very nice indeed. I recently tried to do something very similar with inline SVG sprites using gulp-svg-sprite. The solution I came with was to use the gulp-svg-sprite plugin to save the sprite on a variable (ie.: svgSprite) using the plugins transform and later on, 'inject' it on my page with:

`{ src: svgSprite, tpl: '%s' } : ''`

It works, but it would be better if I could do what @jraoult is suggesting! =)
